MapboxSearch/base/src/test/java/com/mapbox/search/base/location/LocationEngineAdapterTest.kt

```diff
@@ -66,70 +68,34 @@ internal class LocationEngineAdapterTest {
     @TestFactory
     fun `Requests new location if there's no last known location`() = TestCase {
         Given("LocationEngineAdapter with mocked dependencies") {
-            val lastLocationCallbackSlot = slot< LocationEngineCallback<LocationEngineResult>>()
-            every { engine.getLastLocation(capture(lastLocationCallbackSlot)) } answers {
-                lastLocationCallbackSlot.captured.onSuccess(LocationEngineResult.create(emptyList()))
+            val lastLocationCallbackSlot = slot<GetLocationCallback>()
+            every { locationProvider.getLastLocation(capture(lastLocationCallbackSlot)) } answers {
+                lastLocationCallbackSlot.captured.run(null)
+                Cancelable { }
             }
 
-            val locationUpdatesCallbackSlot = slot< LocationEngineCallback<LocationEngineResult>>()
-            every { engine.requestLocationUpdates(any(), capture(locationUpdatesCallbackSlot), any()) } answers {
-                locationUpdatesCallbackSlot.captured.onSuccess(LocationEngineResult.create(createMockedLocation(LOCATION)))
+            val locationUpdatesCallbackSlot = slot<LocationObserver>()
+            every { locationProvider.addLocationObserver(capture(locationUpdatesCallbackSlot)) } answers {
+                locationUpdatesCallbackSlot.captured.onLocationUpdateReceived(listOf(createMockedLocation(LOCATION)))
             }
 
-            adapter = LocationEngineAdapter(app, engine, timeProvider, locationPermissionChecker)
+            adapter = LocationEngineAdapter(app, locationProvider, timeProvider, locationPermissionChecker)
 
             When("LocationEngineAdapter instantiated") {
                 Verify("Location permissions checked") {
                     locationPermissionChecker(app)
                 }
 
                 VerifyOnce("Last known location requested") {
-                    engine.getLastLocation(lastLocationCallbackSlot.captured)
+                    locationProvider.getLastLocation(lastLocationCallbackSlot.captured)
                 }
 
                 VerifyOnce("Location updates requested") {
-                    engine.requestLocationUpdates(any(), locationUpdatesCallbackSlot.captured, any())
+                    locationProvider.addLocationObserver(locationUpdatesCallbackSlot.captured)
                 }
 
                 VerifyOnce("Unsubscribes from location updates when location is received") {
-                    engine.removeLocationUpdates(locationUpdatesCallbackSlot.captured)
-                }
-
-                Then("Adapter returns correct location", LOCATION, adapter.location)
-            }
-        }
-    }
-
-    @TestFactory
-    fun `Requests new location if initial location request failed`() = TestCase {
```

Comment:
What about this test case?

Reply:
`LocationEngineAdapter` now uses `LocationProvider`, and I found no alternative method to check for a failure. So, I didn't see how I could rework this test to make it work.